source/elements/oneMKL/source/domains/blas/omatcopy.rst

```diff
@@ -0,0 +1,316 @@
+.. SPDX-FileCopyrightText: 2022 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_blas_omatcopy:
+
+omatcopy
+========
+
+Computes an out-of-place scaled matrix transpose or copy operation
+using a general matrix.
+
+.. _onemkl_blas_omatcopy_description:
+
+.. rubric:: Description
+
+The ``omatcopy`` routine performs an out-of-place scaled
+matrix copy or transposition.
+
+The operation is defined as:
+
+.. math::
+
+      B \leftarrow \alpha * op(A)
+
+where:
+
+op(``X``) is one of op(``X``) = ``X``, or op(``X``) = ``X``\ :sup:`T`, or op(``X``) = ``X``\ :sup:`H`,
+
+``alpha`` is a scalar,
+
+``A`` and  ``B`` are matrices,
+
+``A`` is ``m`` x ``n`` matrix,
+
+``B`` is ``m`` x ``n`` matrix if ``op`` is non-transpose and an ``n`` x ``m`` matrix otherwise.,
+
+``omatcopy`` supports the following precisions:
+
+   .. list-table::
+      :header-rows: 1
+
+      * -  T 
+      * -  ``float`` 
+      * -  ``double`` 
+      * -  ``std::complex<float>`` 
+      * -  ``std::complex<double>`` 
+
+.. _onemkl_blas_omatcopy_buffer:
+
+omatcopy (Buffer Version)
+-------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::column_major {
+       void omatcopy(sycl::queue &queue,
+                     oneapi::mkl::transpose trans,
+                     std::int64_t m,
+                     std::int64_t n,
+                     T alpha,
+                     sycl::buffer<T, 1> &a,
```

Comment:
does it make sense t ochange this to a=matrix_in  and b = matrix_out  with ld_in and ld_out as well ?

Reply:
Using `a`, `b`, `lda`, `ldb` is extremely common throughout the BLAS spec so I'm inclined to leave it as is. We changed it in `imatcopy` because there we had some genuine confusion about what the meaning was, here I think it is much clearer. I guess the question is if we want this to be consistent with `imatcopy` or consistent with all the other BLAS and BLAS-like routines? I vote for the latter, leaving `imatcopy` as a special case.

Reply:
I vote to be consistent with BLAS domain as well - since these APIs are now included in BLAS headers/documentation.